repos/system_upgrade/el7toel8/actors/sssdcheck/actor.py

```diff
@@ -0,0 +1,98 @@
+    - ldap_groups_use_matching_rule_in_chain was removed and will be ignored
+    - ldap_initgroups_use_matching_rule_in_chain was removed and will be ignored
+    - ldap_sudo_include_regexp changed default from true to false
+    '''
+    
+    name = 'sssd_check'
+    consumes = ()
+    produces = (Report,)
+    tags = (IPUWorkflowTag, ChecksPhaseTag)
+    
+    def process(self):
+        try:
+            self.config = ConfigParser.RawConfigParser()
+            self.config.read('/etc/sssd/sssd.conf')
+        except ConfigParser.Error:
+            # SSSD is not configured properly. Nothing to do.
+            return
+        
+        affected_domains = []
+        
+        for section in self.config.sections():
+            # We are interested only in domains.
+            if not section.startswith('domain/'):
+                continue
+            
+            domain = section[len('domain/'):]
+            self.checkLocalProvider(domain)
+            self.checkRemovedOption(domain, 'ldap_groups_use_matching_rule_in_chain')
+            self.checkRemovedOption(domain, 'ldap_initgroups_use_matching_rule_in_chain')
+            self.checkSudoRegexp(domain)
+
+    def checkLocalProvider(self, domain):
+        provider = self.get_provider(domain, 'id_provider')
+        if provider != 'local':
+            return
+        
+        report_generic(
+            title='SSSD Domain "%s": local provider is no longer '
+                  'supported and the domain will be ignored.' % domain,
+            summary='Local provider is no longer supported.'
+        )
+    
+    def checkRemovedOption(self, domain, option):
+        section = self.get_domain_section(domain)
+        if not self.config.has_option(section, option):
+            return
+        
+        report_generic(
+            title='SSSD Domain "%s": option %s has no longer ' 
+                  'any effect' % (domain, option),
+            summary='Option %s was removed and it will be ignored.' % option
+        ) 
+
+    def checkSudoRegexp(self, domain):
+        section = self.get_domain_section(domain)
+        provider = self.get_provider(domain, 'sudo_provider', ['id_provider'])
+        
+        if provider not in ['ad', 'ldap']:
+            return
+
+        if not self.config.has_option(section, 'ldap_sudo_include_regexp'):
+            return
+        
+        report_generic(
+            title='SSSD Domain "%s": sudo rules containing wildcards ' 
+                  'will stop working.' % domain,
+            summary='Default value of ldap_sudo_include_regexp changed '
+                    'from true to false for performance reason. If you use '
+                    'sudo rules with wildcards, set this option to true '
+                    'explicitly.'
+        )
+        
+        # Produce report
+
+    def get_domain_section(self, domain):
+        return 'domain/' + domain
+
+    def get_provider(self, domain, provider, fallbacks=[]):
```

Comment:
Please, you rather `list()` here.

```suggestion
    def get_provider(self, domain, provider, fallbacks=list()):
```